src/docs/5.31.x/core-development-concepts/basics/routes-and-events.mdx

```diff
@@ -0,0 +1,245 @@
+            method: "POST"
+        });
+    }
+});
+
+```
+
+Next thing you need to do is to add it into the `apps/api/graphql/src/index.ts` file via the `RoutePlugin` or `createApiGatewayRoute` method:
+
+```ts apps/api/graphql/src/index.ts
+// ... other imports
+import { createApiGatewayHandler as createHandler, createApiGatewayRoute } from "@webiny/handler-aws";
+
+export const handler = createHandler({
+    plugins: [
+        // ... other plugins
+        createApiGatewayRoute(({ onPost }) => {
+            onPost("/webiny", async (request, reply) => {
+                // we can log the whole request body
+                console.log(request.body);
+
+                // and we can send some reply
+                reply
+                    .headers({
+                        "x-route-example": "yes"
+                    })
+                    .send({
+                        everything: {
+                            ok: true
+                        }
+                    });
+            });
+        })
+    ],
+    http: { debug }
+});
+```
+
+<Alert type="warning">
+  Security checks will be done as on the <strong>/graphql</strong> endpoint. If you need customizations, feel free to add them.
+</Alert>
+
+<Alert type="danger">
+  Routes can be added only when initializing the handler.
+</Alert>
+
+## Adding Event Handler
+
+The idea behind all our predefined event handlers is to handle events which are not of API Gateway Event type.
+
+Yes, you can catch API Gateway event like this, but there is no point since we have a handler specifically designed for that.
+Of course, feel free to use what ever is available and can help you to achieve your goal.
+
+
+### Example
+Let's say you created a part of code which sends out an SQS message, and you want to have a Lambda which handles that message.
+
+Good example would be if you want to run some calculation, asynchronously, from our GraphQL Lambda.
+You would insert an SQS Message and in turn it would trigger a Lambda which you have defined. That lambda should have code similar to this:
+
+```ts
+import { createSQSHandler, createSQSEventHandler } from "@webiny/handler-aws";
+
+const handler = createSQSHandler({
+  plugins: [
+      createSQSEventHandler(async({request, reply, event, context, lambdaContext}) => {
+          // "context" variable is the same as in our system - of course, if you loaded all the plugins as we do
+          // because it is an sqs event, you know the type of the "event" variable and you can handle it from there
+          const result = await someHeavyCalculation(event);
+          
+          // maybe store that result into the database?
+          await storeResult(result);
+          
+          return reply.send({
+              ok: true
+          });
+      })
+  ]
+})
+```
+
+<Alert type="warning">
+  Note that we do not actually check if the event really is an SQS event. It is expected that you initialize correct handler for event you are expecting.
```

Comment:
```suggestion
  Note that we do not actually check if the event really is an SQS event. It is expected that you initialize the correct handler for event you are expecting. If you add the wrong event it will not execute.
```